### ISSUE TITLE
[Cart] Table element is missing accessible name

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#583]
## Issue summary

The cart `table` element is missing an accessible name. The inclusion of an accessible name helps to convey purpose to page structure when navigating via specific element type alone.

<details>
  <summary>Screenshots</summary>

[Screenshot removed]

</details>

#### HTML

```html
<table class="cart-items">
  <thead>
    <!-- Table content -->
  </thead>
  <tbody>
    <!-- Table content -->
  </tbody>
</table>
```

## Steps to reproduce

1. Go to a Dawn store with any supported browser
2. Add an item to the cart and proceed to the cart page
3. With an active screen reader, navigate via table elements only

### Expected

- Cart `table` to be labelled

### Actual

- Cart `table` is unlabelled

## Recommendation

Add a `caption.visually-hidden` element to the `table`, setting its content to match the `h1` heading. This will provide context on the table purpose while navigating by tables.

#### HTML

```html
<table class="cart-items">
  <caption class="visually-hidden">Your cart</caption>
  <thead>
    <!-- Table content -->
  </thead>
  <tbody>
    <!-- Table content -->
  </tbody>
</table>
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Dawn version:** 2.0
- **Browser:** Any
- **Component:** Cart
- **WCAG Principal:** Perceivable
- **WCAG SC:** [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
- **Severity:** Low
- **Effort:** Low
